### PR TITLE
Switch git submodules to Wally packages

### PR DESCRIPTION
**Pull request by boatbomber**
Removes the git submodules and moves to a more elegant solution with [Wally](https://github.com/UpliftGames/wally/).
Also migrates to @evaera's Promise. No compatibility issues arose, and plugin tests all pass.

### CODE REVIEW

**Review by mkargus — COMMENTED** [review]
Everything looks good.

One minor suggestion, remove the step for submodules on the workflows since the plugin part was the only one needing it:
```diff
  uses: action/checkout@v3
- with:
-    submodules: true
```

**Review by imacodr — APPROVED** [review]
Everything looks good for release to me.

File: plugin/default.project.json [hunk 1]
```diff
@@ -1,33 +1,25 @@
...
+		"Http": {
+			"$path": "http"
+		},
+		"Fmt": {
+			"$path": "fmt"
+		},
+		"RbxDom": {
+			"$path": "rbx_dom_lua"
+		}
+	}
```
Comment (mkargus) [inline plugin/default.project.json:23]: Indents are not consistent.

**Review by mkargus — APPROVED** [review]

File: plugin/src/Reconciler/Error.lua [hunk 2]
```diff
@@ -2,7 +2,8 @@
 	Defines the errors that can be returned by the reconciler.
 ]]
 
-local Fmt = require(script.Parent.Parent.Parent.Fmt)
+local Packages = script.Parent.Parent.Parent.Packages
```
Comment (LPGhatguy) [inline plugin/src/Reconciler/Error.lua:5]: This kinda line is why we use `FindFirstAncestor("Rojo")` a lot. 😅

**Review by LPGhatguy — APPROVED** [review]
Thank you, this is a great migration!

**Pull request merged**